lib/player.js

```diff
@@ -1868,12 +1868,24 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     // Pick the initial streams to play.
     const initialVariant = this.chooseVariant_();
     goog.asserts.assert(initialVariant, 'Must choose an initial variant!');
-    this.switchVariant_(initialVariant, /* fromAdaptation= */ true,
-        /* clearBuffer= */ false, /* safeMargin= */ 0);
+
+    // However, we would skip switch to initial variant
+    // if user already pick variant track (via selectVariantTrack api)
+    const activeVariantTrack = this.getVariantTracks().find((t) => t.active);
+
+    if (!activeVariantTrack) {
+      this.switchVariant_(initialVariant, /* fromAdaptation= */ true,
+          /* clearBuffer= */ false, /* safeMargin= */ 0);
+    }
 
     // Decide if text should be shown automatically.
     const initialTextStream = this.chooseTextStream_();
-    if (initialTextStream) {
+
+    // Similar to video/audio track, we would skip switch initial text track
+    // if user already pick text track (via selectTextTrack api)
+    const activeTextTrack = this.getTextTracks().find((t) => t.active);
+
+    if (!activeTextTrack && initialTextStream) {
```

Comment:
This condition should cover not only addTextStreamToSwitchHistory_(), but also chooseTextStream_() above and setInitialTextState_() and switchTextStream() below.